code/src/pgsql_io/pgsql_io.ml

```diff
@@ -991,7 +1176,19 @@ and create_sm_process_login_frames ?passwd ~user t =
   | AuthenticationSCMCredential :: _ ->
       Abb.Future.return (Error `Unsupported_auth_scm_credential_err)
   | AuthenticationGSS :: _ -> Abb.Future.return (Error `Unsupported_auth_gss_err)
-  | AuthenticationSASL _ :: _ -> Abb.Future.return (Error `Unsupported_auth_sasl_err)
+  | AuthenticationSASL { auth_mechanisms } :: fs
+    when CCList.mem ~eq:CCString.equal "SCRAM-SHA-256" auth_mechanisms -> (
+      match passwd with
+      | Some password ->
+          let open Abbs_future_combinators.Infix_result_monad in
+          let open Auth_scram in
```

Comment:
Arbitrary rule but only open modules where you require operators from it, and be very sparing with opens.

This is because when we read code we want to be able to, as easily as possible, understand where the symbols are coming from.  It also reduces the chances of accidentally shadowing a value or some refactoring having unintentional consequences (maybe we add a value to a module that ends up shadowing another value that we didn't expect.  General heuristic: explicit is better than implicit, but with-in reason.